### PR TITLE
Refactor Server.js

### PR DESCRIPTION
**Pull request by hiroppy**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [x] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

no

### Motivation / Use-Case

`Server.js` has more than 1000 lines and it is hard to test.
So, we split the code.

### Breaking Changes

no

### CODE REVIEW

File: lib/Server.js [hunk 1]
```diff
@@ -201,349 +198,22 @@ class Server {
...
-      },
-    };
-
+    const features = createFeatures.call(this, {
+      app,
+      compiler,
+      options,
+      contentBase,
+      websocketProxies,
+    });
```
Comment (evilebottnawi) [inline lib/Server.js]: Maybe better do this as method `setupFeatures` (we need `this` in context so better use SOLID practice here and do this as method)

**Comment by hiroppy** [comment]
@evilebottnawi PTAL 🙏 https://github.com/webpack/webpack-dev-server/pull/1839/commits/d30d2621692031474e420047463fcccdfed45877

File: lib/Server.js (hunk 1 above)
Comment (evilebottnawi) [reply lib/Server.js]: Apparently I didn’t explain well :smile: I meant to do method `Server.setupFeatures` like:
```js
class Server {
  setupFeatures() {
    // Logic here
  }
}
```

Comment (hiroppy) [reply lib/Server.js]: like this?

```javascript
    const features = this.setupFeatures({
      app,
      compiler,
      options,
      contentBase,
      websocketProxies,
    });

...

  setupFeatures(...args) {
    return createFeatures.call(this, args);
  }

```

Comment (evilebottnawi) [reply lib/Server.js]: put code from `setupFeatures` inside method (SOLID), we use this and using `call` with `this` is not good idea

Comment (hiroppy) [reply lib/Server.js]: hmm... but `createFeatures` has more than 280 lines... and L591 is already used  `call`.

Comment (hiroppy) [reply lib/Server.js]: ok, fixed https://github.com/webpack/webpack-dev-server/pull/1839/commits/1ea70ce11d777388895cba111e51ee59f3d607c1

**Comment by evilebottnawi** [comment]
Good job!

**Pull request merged**